pkg/kubectl/generate/versioned/run.go

```diff
@@ -329,6 +350,15 @@ func getName(params map[string]string) (string, error) {
 	return name, nil
 }
 
+// getNamespace returns the namespace of newly created resource.
+func getNamespace(params map[string]string) (string, error) {
+	namespace, found := params["namespace"]
+	if !found || len(namespace) == 0 {
+		return "", fmt.Errorf("'namespace' is a required parameter")
```

Comment:
this breaks compatibility with existing invocations. if `--namespace` is omitted, continue outputting objects with no namespace set

Reply:
thanks. Agree.
PR updated, please help take another look, thanks. :)